simvue/client.py

```diff
@@ -291,11 +292,32 @@ def save(self, filename, category):
         if not os.path.isfile(filename):
             raise RuntimeError('File %s does not exist' % filename)
 
+        if filetype:
+            mimetypes_valid = []
+            mimetypes.init()
+            for item in mimetypes.types_map:
+                mimetypes_valid.append(mimetypes.types_map[item])
+
+            if filetype not in mimetypes_valid:
+                raise RuntimeError('Invalid MIME type specified')
+
         data = {}
         data['name'] = os.path.basename(filename)
         data['run'] = self._name
         data['category'] = category
         data['checksum'] = calculate_sha256(filename)
+        data['size'] = os.path.getsize(filename)
```

Comment:
Do we need info on the location of the file? If there are a lot of files, will save time tracking it down?

Reply:
Will also be useful to have a couple of tests. I can add this.

Reply:
> Do we need info on the location of the file? If there are a lot of files, will save time tracking it down?

We're using object storage using the checksum as the name of the object. This means files are de-duplicated and any file can be obtained easily, but this is all hidden from users.

An extension could be allow users to store files in arbitrary storage systems. In this case it would be necessary to store locations. However, I prefer not to go down that road (at least to begin, unless we find a good reason to do so).

Reply:
Or do you mean location of where the file was originally? I guess that could actually be helpful to users, hadn't thought of that.

Reply:
Yes, as there will be a lot of files to keep track of, will be useful to have this info stored as reference, if users wanted to check. So perhaps add it as another input?

Reply:
Maybe `data['path']` or `data['originalPath']`? (Since this will be JSON going to a REST API my assumption is that we should use the Google JSON Style Guide, not Python conventions).

Reply:
Yes. I would suggest data['originalPath'] (to be explicit) and going by the assumption of catering to generic sim users (C++, FORTRAN etc) I would think - Google JSON Style Guide

Reply:
Added `originalPath`